recipe-app/ViewModel/RecipeListViewModel.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  RecipeListViewModel.swift
+//  recipe-app
+//
+//  Created by Arthur Hermann on 27/11/2024.
+//
+
+import Foundation
+import UIKit
+
+ class RecipeListViewModel{
+    let key = "5b91aa819e6d4bc8848f4c972103e6dc"
+    var recipes = [Recipe]()
+    
+     /*
+      Retrieves first 20 recipes matching the user's search.
+      */
+     func searchRecipe(_ query: String) {
+         guard let url = URL(string: "https://api.spoonacular.com/recipes/complexSearch?apiKey=\(key)&query=\(query)&number=20") else {return}
```

Comment:
Instead of hardcoding the URL consider creating a method to make it more reusable and generic. 

```
    private func makeURL(query: String) -> URL? {
        var components = URLComponents(string: "https://api.spoonacular.com/recipes/complexSearch")
        components?.queryItems = [
            URLQueryItem(name: "apiKey", value: apiKey),
            URLQueryItem(name: "query", value: query),
            URLQueryItem(name: "number", value: "20")
        ]
        return components?.url
    }
```